Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -121,6 +124,32 @@ static const AtomString& attachmentSubtitleIdentifier()
     return identifier;
 }
 
+static const AtomString& attachmentSaveIdentifier()
+{
+    static MainThreadNeverDestroyed<const AtomString> identifier("attachment-save"_s);
+    return identifier;
+}
+
+static QualifiedName saveQualifiedName()
+{
+    return QualifiedName { nullAtom(), "save"_s, nullAtom() };
+}
+
+void HTMLAttachmentElement::addShadowElement(RefPtr<HTMLElement> element, const AtomString& id, String innerText)
+{
+    element->setIdAttribute(id);
+    if (!innerText.isEmpty())
+        element->setInnerText(WTFMove(innerText));
```

Comment:
nit: `setTextContent()` (`.textContent =` in JS) is faster if you're not relying on innerText specific behaviors.